js/screens/Activity/Activity.js

```diff
@@ -1,34 +1,36 @@
 import React from 'react';
 import { View, Text, ScrollView } from 'react-native';
 import styles from './styles';
+import PropTypes from 'prop-types';
 
-const Activity = props => {
+const Activity = ({ data }) => {
+  const activity = data.Activity;
   return (
-    <View style={styles.container}>
-      <Text style={styles.header}>The Marshmallow Challenge</Text>
-      <Text style={styles.title}>STEM Challenge:</Text>
-      <Text style={styles.description}>
-        Build the tallest tower possible utilizing the materials
-      </Text>
-      <Text style={styles.title}>Procedures:</Text>
-      <View>
-        <Text style={styles.procedure}>
-          1. Give materials to each group. Each group can use 5 feet of masking
-          tape, 25 pieces of spaghetti, 5 feet of string, 10 popsicle sticks,
-          and 2 marshmallows. The educator may modify the material list as
-          desired.
-        </Text>
-        <Text style={styles.procedure}>
-          2. Set a time limit of 20 minutes, and begin the challenge!
-        </Text>
-        <Text style={styles.procedure}>
-          3. Monitor groups as they work. The educator may choose to ask a few
-          reflective questions such as “what strategy are you using?” or “what
-          have you tried so far, and how are you improving your design?”
-        </Text>
+    <ScrollView>
+      <View style={styles.container}>
+        <Text style={styles.title}>{activity.title}</Text>
+        <Text style={styles.introText}>STEM Challenge:</Text>
+        <Text style={styles.description}>{activity.description}</Text>
+        <Text style={styles.introText}>Procedures:</Text>
+        {activity.procedures.map((procedure, index) => (
+          <View key={index} style={styles.procedureBlock}>
+            <Text>{(index += 1)}.</Text>
+            <Text style={styles.procedureText}>{procedure}</Text>
+          </View>
+        ))}
       </View>
-    </View>
+    </ScrollView>
   );
 };
 
+Activity.propTypes = {
+  data: PropTypes.shape({
+    Activity: PropTypes.shape({
```

Comment:
Is your data shape of some other shape? If not then please remove Activity shape.

Reply:
The Activity component accepts props of "data" that has an object "Activity" inside it.